src/lib/djerba/helpers/expression_helper/helper.py

```diff
@@ -0,0 +1,122 @@
+from djerba.util.subprocess_runner import subprocess_runner
+
+class main(helper_base):
+
+    ENSCON_KEY = 'enscon'
+    GENE_LIST_KEY = 'gene_list'
+    GEP_REFERENCE_KEY = 'gep_reference'
+    RSEM_GENES_RESULTS_KEY = 'rsem_genes_results'
+    TCGA_DATA_KEY = 'tgca_data'
+    TCGA_CODE_KEY = 'tgca_code'
+    TUMOUR_ID = 'tumour_id' # TODO put in core params
+
+    # 0-based index for GEP results file
+    GENE_ID = 0
+    FPKM = 6
+    
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        defaults = {
+            self.ENSCON_KEY: '${DJERBA_DATA_DIR}/ensemble_conversion_hg38.txt',
+            self.GENE_LIST_KEY: '${DJERBA_DATA_DIR}/targeted_genelist.txt',
+            self.GEP_REFERENCE_KEY: '/home/ibancarz/workspace/djerba/test/20230505_02/gep_reference.txt.gz', # TODO FIXME
+            self.RSEM_GENES_RESULTS_KEY: 'NULL',
+            self.TCGA_DATA_KEY: '/.mounts/labs/CGI/gsi/tools/RODiC/data',
+            self.TCGA_CODE_KEY: 'NULL'
+        }
+        self.set_all_ini_defaults(defaults)
+        self.add_ini_required(self.TCGA_CODE_KEY)
+
+    def configure(self, config):
+        config = self.apply_defaults(config)
+        wrapper = self.get_config_wrapper(config)
+        wrapper.apply_my_env_templates()
+        fpr_path = self.workspace.abs_path(fpr_helper.PROVENANCE_OUTPUT)
+        donor = wrapper.get_core_string(core_constants.DONOR)
+        project = wrapper.get_core_string(core_constants.PROJECT)
+        if wrapper.get_my_string(self.TCGA_CODE_KEY) == 'NULL': # TODO implement is_null()
+            wrapper.set_my_param(self.TCGA_CODE_KEY, project)
+        samples = sample_name_container() # empty container; TODO sample names in INI
+        reader = provenance_reader(fpr_path, project, donor, samples)
+        rsem_genes_results = reader.parse_gep_path()
+        wrapper.set_my_param(self.RSEM_GENES_RESULTS_KEY, rsem_genes_results)
+        return wrapper.get_config()
+
+    def extract(self, config):
+        wrapper = self.get_config_wrapper(config)
+        results = wrapper.get_my_string(self.RSEM_GENES_RESULTS_KEY)
+        gep_reference = wrapper.get_my_string(self.GEP_REFERENCE_KEY)
+        tumour_id = wrapper.get_core_string(self.TUMOUR_ID)
+        gep_abs_path = self.preprocess_gep(results, gep_reference, tumour_id)
+        # Run the R script with output to the workspace
+        cmd = [
+            'Rscript',
+            os.path.join(self.get_module_dir(), 'find_expression.R'),
+            '--enscon ', wrapper.get_my_string(self.ENSCON_KEY),
+            '--genelist', wrapper.get_my_string(self.GENE_LIST_KEY),
+            '--gepfile', gep_abs_path,
+            '--outdir', self.workspace.get_work_dir(),
+            '--tcgadata', wrapper.get_my_string(self.TCGA_DATA_KEY),
+            '--tcgacode', wrapper.get_my_string(self.TCGA_CODE_KEY)
+        ]
+        self.logger.debug("Rscript command: "+" ".join(cmd))
+        #subprocess_runner(self.log_level, self.log_path).run(cmd)
```

Comment:
- Not running the R script yet
- I need to configure my test environment with some R prerequisites first
- Also, the R script itself will likely need some debugging